src/thor/route_action.cc

```diff
@@ -416,6 +416,9 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
         // Form output information based on path edges
         if (api.trip().routes_size() == 0 || api.options().alternates() > 0)
           route = api.mutable_trip()->mutable_routes()->Add();
+        if (route == nullptr) {
```

Comment:
this is pedantic. the only way this can happen is if you run your system out of ram. if that happens you have way worse problems on your  hands. id prefer we not add this check